code/modules/mob/mob_helpers.dm

```diff
@@ -479,3 +479,25 @@
 	else if(iscarbon(mob) || isAI(mob) || isbrain(mob))
 		divided_health = abs(HEALTH_THRESHOLD_DEAD - mob.health) / abs(HEALTH_THRESHOLD_DEAD - mob.maxHealth)
 	return divided_health * 100
+
+/**
+ * Generates a log message when a user manually changes their targeted zone.
+ * Only need to one of new_target or old_target, and the other will be auto populated with the current selected zone.
+ */
+/mob/proc/log_manual_zone_selected_update(source, new_target, old_target)
+	if(!new_target && !old_target)
+		CRASH("Called log_manual_zone_selected_update without specifying a new or old target")
+
+	old_target ||= zone_selected
+	new_target ||= zone_selected
+	if(old_target == new_target)
+		return
+
+	var/atom/handitem = get_active_held_item()
+	var/atom/offhand = get_inactive_held_item()
+	var/log_message = "[key_name(src)] changed selected zone from [old_target] to [new_target]."
+	if(handitem)
+		log_message += " active item: ([handitem.type])[handitem]"
+	if(offhand)
+		log_message += " inactive item: ([offhand.type])[offhand]"
+	log_game(log_message)
```

Comment:
This seems like it could get pretty spammy, why put this in game log and not its own log?

Reply:
In its own log is fine I just told them this because it means we don't have to worry about getting it on parsed-logs, easy enough to do that though